server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationAllocationIT.java

```diff
@@ -0,0 +1,1204 @@
+                    .put("index.routing.allocation.exclude._name", allNodesExcept(targetNodeName))
+            )
+            .execute()
+            .actionGet();
+
+        ensureGreen(TEST_INDEX);
+
+        ClusterRerouteResponse rerouteResponse = client.admin()
+            .cluster()
+            .prepareReroute()
+            .setExplain(true)
+            .add(new MoveAllocationCommand(TEST_INDEX, 0, currentNodeName, targetNodeName))
+            .execute()
+            .actionGet();
+
+        ensureGreen(TEST_INDEX);
+
+        return rerouteResponse;
+    }
+
+    // verify that shard does not exist at targetNode
+    private void assertNonAllocation(ShardRouting shardRouting) {
+        assertFalse(shardRouting.active());
+        assertNull(shardRouting.currentNodeId());
+        assertEquals(ShardRoutingState.UNASSIGNED, shardRouting.state());
+    }
+
+    // verify that shard exists at targetNode
+    private void assertAllocation(ShardRouting shardRouting, DiscoveryNode targetNode) {
+        assertTrue(shardRouting.active());
+        assertNotNull(shardRouting.currentNodeId());
+        assertEquals(shardRouting.currentNodeId(), targetNode.getId());
+    }
+
+    // verify that shard did not get relocated to the targetNode
+    private void assertNonRelocation(
+        ShardRouting shardRouting,
+        ClusterRerouteResponse rerouteResponse,
+        DiscoveryNode currentNode,
+        DiscoveryNode targetNode,
+        String reason
+    ) {
+        Decision.Type decisionType = rerouteResponse.getExplanations().explanations().get(0).decisions().type();
+        List<Decision> relocationDecisions = rerouteResponse.getExplanations().explanations().get(0).decisions().getDecisions();
+        for (Decision dec : relocationDecisions) {
+            if (dec.type().equals(Decision.Type.NO)) {
+                // only one NO decision
+                assertEquals(reason, dec.getExplanation());
+            }
+        }
+        ShardRouting expectedShardRouting = rerouteResponse.getState()
+            .getRoutingTable()
+            .index(TEST_INDEX)
+            .shard(0)
+            .shards()
+            .stream()
+            .filter(sr -> sr.allocationId().equals(shardRouting.allocationId()))
+            .findFirst()
+            .orElse(null);
+        assertNotNull(expectedShardRouting);
+        assertEquals(Decision.Type.NO, decisionType);
+        assertNotNull(expectedShardRouting.currentNodeId());
+        assertEquals(currentNode.getId(), expectedShardRouting.currentNodeId());
+        assertNotEquals(targetNode.getId(), expectedShardRouting.currentNodeId());
+    }
+
+    // index settings builder for remote store enabled index
+    private Settings.Builder getRemoteStoreBackedIndexSettingsBuilder(boolean isRemoteStoreBackedIndex) {
+        Settings.Builder builder = Settings.builder().put(SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT);
```

Comment:
You don't need this as cluster setting would automatically populate this . But it would need `index settings` PR which you are already working upon to correctly populate it.  For now , you can  put these ITs in that PR only  . UTs should be good enough for getting it in .